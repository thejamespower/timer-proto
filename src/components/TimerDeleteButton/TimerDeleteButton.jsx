import CardActions from '@material-ui/core/CardActions/CardActions'
import Button from '@material-ui/core/Button/Button'
import DeleteIcon from '@material-ui/icons/Delete'
import React from 'react'
import PropTypes from 'prop-types'

const TimerDeleteButton = ({
  active, complete, superTimerActive, deleteTimer, id,
}) => ((active || complete || superTimerActive) ? null : (
  <div>
    <CardActions>
      <Button onClick={() => deleteTimer(id)}>Delete <DeleteIcon /></Button>
    </CardActions>
  </div>
))

TimerDeleteButton.propTypes = {
  active: PropTypes.bool.isRequired,
  complete: PropTypes.bool.isRequired,
  superTimerActive: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  deleteTimer: PropTypes.func.isRequired,
}

export default TimerDeleteButton
