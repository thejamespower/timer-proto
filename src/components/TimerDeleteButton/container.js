import { connect } from 'react-redux';

import { bindActionCreators } from 'redux';
import TimerDeleteButton from './TimerDeleteButton';
import { deleteTimer } from '../../action-creators';

const mapStateToProps = null;

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      deleteTimer,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(TimerDeleteButton);
