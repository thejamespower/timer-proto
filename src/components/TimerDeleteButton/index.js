import TimerDeleteButton from './container'

export default TimerDeleteButton
