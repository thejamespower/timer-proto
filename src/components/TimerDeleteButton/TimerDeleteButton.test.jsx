import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from '@material-ui/core/Button/Button';

import TimerDeleteButton from './TimerDeleteButton';

describe('TimerDeleteButton', () => {
  let props;

  beforeEach(() => {
    props = {
      active: false,
      complete: false,
      superTimerActive: false,
      deleteTimer: () => null,
      id: 'test',
    };
  });

  describe('render', () => {
    it('renders default state', () => {
      const component = renderer.create(<TimerDeleteButton {...props} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders active state', () => {
      props.active = true;

      const component = renderer.create(<TimerDeleteButton {...props} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders complete state', () => {
      props.complete = true;

      const component = renderer.create(<TimerDeleteButton {...props} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders superTimerActive state', () => {
      props.superTimerActive = true;

      const component = renderer.create(<TimerDeleteButton {...props} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('behaviour', () => {
    it('handles button click', () => {
      props.deleteTimer = jest.fn();

      const wrapper = shallow(<TimerDeleteButton {...props} />);
      const button = wrapper.find(Button);
      button.prop('onClick')();

      expect(props.deleteTimer).toHaveBeenCalledWith(props.id);
    });
  });
});
