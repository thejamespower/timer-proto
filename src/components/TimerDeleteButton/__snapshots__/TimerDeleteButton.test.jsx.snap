// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerDeleteButton render renders active state 1`] = `null`;

exports[`TimerDeleteButton render renders complete state 1`] = `null`;

exports[`TimerDeleteButton render renders default state 1`] = `
<div>
  <div
    className="MuiCardActions-root-1"
  >
    <button
      className="MuiButtonBase-root-30 MuiButton-root-4 MuiButton-text-6 MuiButton-flat-9 MuiCardActions-action-3"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-5"
      >
        Delete 
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-33"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-42"
      />
    </button>
  </div>
</div>
`;

exports[`TimerDeleteButton render renders superTimerActive state 1`] = `null`;
